File: app/src/main/java/com/iita/akilimo/rest/request/RecommendationRequest.java
package com.iita.akilimo.rest.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.iita.akilimo.models.Fertilizer;

import java.util.List;

import lombok.Data;

@Data
public class RecommendationRequest {

    @JsonProperty("computeRequest")
    private ComputeRequest computeRequest;

    @JsonProperty("fertilizerList")
    private List<Fertilizer> fertilizerList;
}

File: app/src/main/java/com/iita/akilimo/rest/request/RecommendationRequest.kt
package com.iita.akilimo.rest.request

import com.fasterxml.jackson.annotation.JsonProperty
import com.iita.akilimo.entities.Fertilizer

class RecommendationRequest(
    @JsonProperty("userInfo")
    var userInfo: UserInfo,

    @JsonProperty("computeRequest")
    var computeRequest: ComputeRequest,

    @JsonProperty("fertilizerList")
    var fertilizerList: List<Fertilizer>
)
